tests/logprob/test_mixture.py

```diff
@@ -927,40 +929,73 @@ def test_switch_mixture():
-
-    assert equal_computations(fgraph.outputs, fgraph2.outputs)
-
     z1_logp = conditional_logp({Z1_rv: z_vv, I_rv: i_vv})
     z2_logp = conditional_logp({Z2_rv: z_vv, I_rv: i_vv})
     z1_logp_combined = pt.sum([pt.sum(factor) for factor in z1_logp.values()])
     z2_logp_combined = pt.sum([pt.sum(factor) for factor in z2_logp.values()])
-
-    # below should follow immediately from the equal_computations assertion above
-    assert equal_computations([z1_logp_combined], [z2_logp_combined])
-
     np.testing.assert_almost_equal(0.69049938, z1_logp_combined.eval({z_vv: -10, i_vv: 1}))
     np.testing.assert_almost_equal(0.69049938, z2_logp_combined.eval({z_vv: -10, i_vv: 1}))
 
 
+@pytest.mark.parametrize("switch_cond_scalar", (True, False))
+def test_switch_mixture_vector(switch_cond_scalar):
+    if switch_cond_scalar:
+        switch_cond = pt.scalar("switch_cond", dtype=bool)
+    else:
+        switch_cond = pt.vector("switch_cond", dtype=bool)
+    true_branch = pt.exp(pt.random.normal(size=(4,)))
+    false_branch = pt.abs(pt.random.normal(size=(4,)))
+
+    switch = pt.switch(switch_cond, true_branch, false_branch)
+    switch.name = "switch_mix"
+    switch_value = switch.clone()
+    switch_logp = logp(switch, switch_value)
+
+    if switch_cond_scalar:
+        test_switch_cond = np.array(0, dtype=bool)
+    else:
+        test_switch_cond = np.array([0, 1, 0, 1], dtype=bool)
+    test_switch_value = np.linspace(0.1, 2.5, 4)
+    np.testing.assert_allclose(
+        switch_logp.eval({switch_cond: test_switch_cond, switch_value: test_switch_value}),
+        np.where(
+            test_switch_cond,
+            logp(true_branch, test_switch_value).eval(),
+            logp(false_branch, test_switch_value).eval(),
+        ),
+    )
+
+
+def test_switch_mixture_invalid_bcast():
+    """Test that we don't mark switches where components are broadcsated as measurable"""
+    valid_switch_cond = pt.vector("switch_cond", dtype=bool)
+    invalid_switch_cond = pt.matrix("switch_cond", dtype=bool)
+
+    valid_true_branch = pt.exp(pt.random.normal(size=(4,)))
+    valid_false_branch = pt.abs(pt.random.normal(size=(4,)))
+    invalid_false_branch = pt.abs(pt.random.normal(size=()))
+
+    valid_mix = pt.switch(valid_switch_cond, valid_true_branch, valid_false_branch)
+    fgraph, _, _ = construct_ir_fgraph({valid_mix: valid_mix.type()})
+    assert isinstance(fgraph.outputs[0].owner.op, MeasurableVariable)
+
+    invalid_mix = pt.switch(invalid_switch_cond, valid_true_branch, valid_false_branch)
+    fgraph, _, _ = construct_ir_fgraph({invalid_mix: invalid_mix.type()})
+    assert not isinstance(fgraph.outputs[0].owner.op, MeasurableVariable)
+
+    invalid_mix = pt.switch(valid_switch_cond, valid_true_branch, invalid_false_branch)
+    fgraph, _, _ = construct_ir_fgraph({invalid_mix: invalid_mix.type()})
+    assert not isinstance(fgraph.outputs[0].owner.op, MeasurableVariable)
```

Comment:
Would it be more clear to rename the `fgraph`s and `invalid_mix`es in this test?

Reply:
sure!

Reply:
On a second thought... rename to what? I feel the names are pretty reasonable?